pydantic/config.py

```diff
@@ -36,183 +34,159 @@ class Extra(str, Enum):
-    validate_assignment: bool = False
-    error_msg_templates: Dict[str, str] = {}  # TODO remove
-    arbitrary_types_allowed: bool = False  # TODO default True, or remove
-    undefined_types_warning: bool = True  # TODO review docs
-    orm_mode: bool = False  # TODO rename to from_attributes
-    alias_generator: Optional[Callable[[str], str]] = None
-    keep_untouched: Tuple[type, ...] = ()  # TODO remove??
-    schema_extra: Union[Dict[str, Any], 'SchemaExtraCallable'] = {}  # TODO remove, new model method
-    json_loads: Callable[[str], Any] = json.loads  # TODO decide
-    json_dumps: Callable[..., str] = json.dumps  # TODO decide
-    json_encoders: Dict[Union[Type[Any], str, ForwardRef], Callable[..., Any]] = {}  # TODO decide
-    underscore_attrs_are_private: bool = False  # TODO remove
-    allow_inf_nan: bool = True
-
-    strict: bool = False
+    strict: bool
 
     # whether inherited models as fields should be reconstructed as base model,
     # and whether such a copy should be shallow or deep
-    copy_on_model_validation: Literal['none', 'deep', 'shallow'] = 'shallow'  # TODO remove???
+    copy_on_model_validation: Literal['none', 'deep', 'shallow']  # TODO remove???
 
-    # whether `Union` should check all allowed types before even trying to coerce
-    smart_union: bool = False  # TODO remove
     # whether dataclass `__post_init__` should be run before or after validation
-    post_init_call: Literal['before_validation', 'after_validation'] = 'before_validation'  # TODO remove
-
-    @classmethod
-    def get_field_info(cls, name: str) -> Dict[str, Any]:
-        """
-        Get properties of FieldInfo from the `fields` property of the config class.
-        """
-
-        fields_value = cls.fields.get(name)
-
-        if isinstance(fields_value, str):
-            field_info: Dict[str, Any] = {'alias': fields_value}
-        elif isinstance(fields_value, dict):
-            field_info = fields_value
-        else:
-            field_info = {}
-
-        if 'alias' in field_info:
-            field_info.setdefault('alias_priority', 2)
-
-        if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
-            alias = cls.alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
-            field_info.update(alias=alias, alias_priority=1)
-        return field_info
-
-    @classmethod
-    def prepare_field(cls, field: Any) -> None:
-        """
-        Optional hook to check or modify fields during model creation.
-        """
-        pass
-
-
-def get_config(config: Union[ConfigDict, Type[object], None]) -> Type[BaseConfig]:
+    post_init_call: Literal['before_validation', 'after_validation']  # TODO remove
+
+    # TODO where to impelement these two classmethods?
+    # @classmethod
+    # def get_field_info(cls, name: str) -> Dict[str, Any]:
```

Comment:
delete it, we're removing `Config.fields`

Reply:
Done